js/flashcards.js

```diff
@@ -568,28 +570,34 @@ H5P.Flashcards = (function ($, XapiGenerator) {
    *   Class to add to existing current card.
    */
   C.prototype.setCurrent = function ($card) {
+    
+    this.$prev = this.$current;
+    
     // Remove from existing card.
     if (this.$current) {
       this.$current.find('.h5p-textinput').attr('tabindex', '-1');
       this.$current.find('.joubel-tip-container').attr('tabindex', '-1');
       this.$current.find('.h5p-check-button').attr('tabindex', '-1');
       this.$current.find('.h5p-icon-button').attr('tabindex', '-1');
     }
-
-    // Set new card
+    
+    // Set new card    
     this.$current = $card;
 
     /* We can't set focus on anything until the transition is finished.
        If we do, iPad will try to center the focused element while the transition
        is running, and the card will be misplaced */
-    $card.one('transitionend', function () {
-      if ($card.hasClass('h5p-current') && !$card.find('.h5p-textinput')[0].disabled) {
-        $card.find('.h5p-textinput').focus();
-      }
-      setTimeout(function () {
-        this.announceCurrentPage();
-      }.bind(this), 500);
-    }.bind(this));
+       
+       // Handle focus shift and screen reader announcement
```

Comment:
Your IDE (if configures properly) should also be able to tell you that the indentation is odd here.